pkg/tbtc/chain.go

```diff
@@ -120,6 +121,58 @@ type DistributedKeyGenerationChain interface {
 	DKGParameters() (*DKGParameters, error)
 }
 
+// InactivityClaimedEvent represents an inactivity claimed event. It is emitted
+// after a submitted inactivity claim lands on the chain.
+type InactivityClaimedEvent struct {
+	WalletID    [32]byte
+	Nonce       *big.Int
+	Notifier    chain.Address
+	BlockNumber uint64
+}
+
+// InactivityChainClaim represents an inactivity claim submitted to the chain.
+type InactivityChainClaim struct {
+	WalletID               [32]byte
+	InactiveMembersIndices []group.MemberIndex
+	HeartbeatFailed        bool
+	Signatures             []byte
+	SigningMembersIndices  []group.MemberIndex
+}
+
+type InactivityClaimChain interface {
+	// OnInactivityClaimed registers a callback that is invoked when an on-chain
+	// notification of the inactivity claim submission is seen.
+	OnInactivityClaimed(
+		func(event *InactivityClaimedEvent),
+	) subscription.EventSubscription
+
+	// AssembleInactivityClaim assembles the inactivity chain claim according to
+	// the rules expected by the given chain.
+	AssembleInactivityClaim(
+		walletID [32]byte,
+		inactiveMembersIndices []group.MemberIndex,
+		signatures map[group.MemberIndex][]byte,
+		heartbeatFailed bool,
+	) (*InactivityChainClaim, error)
+
+	// SubmitInactivityClaim submits the inactivity claim to the chain.
+	SubmitInactivityClaim(
+		claim *InactivityChainClaim,
+		nonce *big.Int,
+		groupMembers []uint32,
+	) error
+
+	// CalculateInactivityClaimSignatureHash calculates hash for the given
+	// inactivity claim.
+	CalculateInactivityClaimSignatureHash(
```

Comment:
```suggestion
	CalculateInactivityClaimHash(
```

Reply:
Done in 0e3b9e5a5e7bfe2b010ecf77509bdfe5536713a2.